packages/react-composites/src/composites/ChatComposite/file-sharing/FileCard.tsx

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { Icon, mergeStyles, ProgressIndicator, Stack, Text, useTheme } from '@fluentui/react';
+import { getFileTypeIconProps } from '@fluentui/react-file-type-icons';
+import React from 'react';
+
+/**
+ * @beta
+ */
+export interface FileCardProps {
+  /**
+   * File name.
+   */
+  fileName: string;
+  /**
+   * Extension of the file used for rendering the file icon.
+   */
+  fileExtension: string;
+  /**
+   * File upload/download progress percentage between 0 and 1.
+   * Not shown if undefined or 1.
+   */
+  progress?: number;
+  /**
+   * Icon to display for actions like download, upload, etc. on the right of file name.
+   */
+  actionIcon?: JSX.Element;
+  /**
+   * Function that runs when actionIcon is clicked
+   */
+  actionHandler?: () => void;
+}
+
+/**
+ * @beta
+ */
+export const FileCard = (props: FileCardProps): JSX.Element => {
+  const { fileName, fileExtension, progress, actionIcon } = props;
+  const theme = useTheme();
+
+  const progressIndicator = (): JSX.Element => (
+    <ProgressIndicator
+      percentComplete={progress}
+      styles={{
```

Comment:
nit: can the styles be constants defined outside the render loop

Reply:
I can do that for sure. However, I don't see the utility in this scenario. Am I missing something?

Reply:
For this component as it stands - not too much utility as presumably re-renders of this component will be negligible, but forcing this as a design pattern everywhere ensures we avoid scenarios where re-renders are expensive. This component could also grow or take in a child component in the future and the unnecessary re-renders could then be impactful.

Also, but a completely personal preference, I find it keeps the code small and functional - for this case once you hoist the styles outside component its clear this component doesn't do anything that the fluent component isn't already doing so means you no longer need to wrap the fluent component and can make this inline instead - making the code more streamlined

Reply:
I can appreciate the personal preference you have.
However, I would like to understand the `re-render` logic here a bit more.
Are we saying that inline styles cause un-necessary re-renders?
OR that calling `mergeStyles` on every re-render is expensive?

Reply:
When this was reviewed it was a raw styles object and not merge styles:
![image](https://user-images.githubusercontent.com/2684369/150890581-4975b208-99ec-44a7-a85a-5e6044271ff7.png)
And yeah raw styles into the styles prop causes the whole component to re-render as it creates a new reference each time https://github.com/microsoft/fluentui/wiki/Performance
Merge styles also works though to prevent the re-rendering instead of extracting constant out

Reply:
Ah yes, I remember now. Thanks for sharing this doc.

Reply:
@prprabhu-ms makes sense to do it for `styles`. For `mergeStyles` I am unable to find a doc on perf impact if the styles are inline.

Reply:
I will do it :D
I just want to understand if there are stronger reasons other than personal preferences.